partner/WHMCS/modules/servers/jetpack/jetpack.php

```diff
@@ -0,0 +1,365 @@
+ * consider the provisioning to be incomplete and get a useful error from the response
+ *
+ * @param array WHMCS $params
+ * @return string Either 'success' or an error with what went wrong when provisioning
+ */
+function jetpack_CreateAccount(array $params)
+{
+
+    $module_errors = validate_required_fields($params);
+    if ($module_errors !== true) {
+        return $module_errors;
+    }
+
+    $access_token = get_access_token($params);
+    if (strpos($access_token, 'JETPACK MODULE') === 0) {
+        return $access_token;
+    }
+
+    $provisioning_url = "https://public-api.wordpress.com/rest/v1.3/jpphp/provision";
+    $stripped_url = preg_replace("(^https?://)", "", $params['customfields']['Site URL']);
+    $stripped_url = rtrim($stripped_url, '/');
+
+    $request_data = [
+        'plan' => strtolower($params['customfields']['Plan']),
+        'siteurl' => $stripped_url,
+        'local_user' => $params['customfields']['Local User'],
+        'force_register' => true,
+    ];
+
+    $response = make_api_request($provisioning_url, $access_token, $request_data);
+    if ($response->success && $response->success == true) {
+        if ($response->next_url) {
+            save_provisioning_details($response->next_url, $params);
+        } elseif (!$response->next_url && $response->auth_required) {
+            save_provisioning_details($response->next_url, $params, true);
+        }
+        return 'success';
+    } else {
+        $errors = get_provisioning_errors_from_response($response);
+        return $errors;
+    }
+}
+
+/**
+ * Equivalent to partner/cancel. Cancel a Jetpack plan using using a Jetpack Hosting partner account. This has
+ * the same prerequiste steps as jetpack_createAccount and will return an error string if the module has not been
+ * setup correctly or there was a failure getting an access token.
+ *
+ * The url scheme for the site being cancelled is not necessary when making this request and is stripped along
+ * with trailing slashes
+ *
+ * If the response json does not contain "success" return error strings based on the response properties
+ *
+ * @param array WHMCS $params
+ * @return string Either 'success' or an error with what went wrong when provisioning
+ */
+function jetpack_TerminateAccount(array $params)
+{
+    $module_errors = validate_required_fields($params);
+    if ($module_errors !== true) {
+        return $module_errors;
+    }
+
+    $access_token = get_access_token($params);
+    if (strpos($access_token, 'JETPACK MODULE') === 0) {
+        return $access_token;
+    }
+
+    $stripped_url = preg_replace("(^https?://)", "", $params['customfields']['Site URL']);
+    $clean_url = rtrim($stripped_url, '/');
+    $clean_url = str_replace('/', '::', $clean_url);
+
+
+    $request_url = 'https://public-api.wordpress.com/rest/v1.3/jpphp/' . $clean_url . '/partner-cancel';
+    $response = make_api_request($request_url, $access_token);
+    if ($response->success === true) {
```

Comment:
What happens if the `success` attribute of `$response` is not set?

Reply:
This was similar to the jetpack_CreateAccount. I've added a check on this one as well to make sure the property is set before continuing.